File: CheckPoints/PrettyPrint.java
public class PrettyPrint
{
    public int[][] prettyPrint(int A) {
        int size = A*2 - 1;

        int[][] matrix = new int[size][size];

        for(int i=0;i<size;i++)
            for(int j=0;j<size;j++)
                matrix[i][j] = 0;

        for( int i=0; i<A; i++ ) {
            int fillValue = A-i;
            int squareSize = fillValue * 2 -1;
            drawSquare( matrix, i, i, squareSize, fillValue );
        }

        return matrix;
    }

    void printMatrix( int[][] matrix ) {
        int size = matrix[0].length;
        for(int i=0;i<size;i++) {
            for(int j=0;j<size;j++)
                System.out.print( matrix[i][j] + " " );
            System.out.println();
        }
    }

    void drawSquare( int[][] matrix, int row, int col, int size, int fillValue ) {
        int end = row + size - 1;
        for( int i=row; i<=end; i++ ) {
            matrix[row][i] = fillValue;
            matrix[end][i] = fillValue;
            matrix[i][col] = fillValue;
            matrix[i][end] = fillValue;
        }
    }
}

File: CheckPoints/PrettyPrint.swift
import Foundation

class Solution {
    func prettyPrint(_ A: inout Int) -> [[Int]] {
        let size = A * 2 - 1
        
        var matrix = Array(repeating: Array(repeating: 0, count: size), count: size)
        for i in 0..<A {
            let fillValue = A-i
            let squareSize = fillValue * 2 - 1
            drawSquare( matrix: &matrix, row: i, col: i, size: squareSize, fillValue: fillValue )
        }
        
        return matrix
    }

    func drawSquare( matrix: inout [[Int]], row: Int, col: Int, size: Int, fillValue: Int ) {
        let end = row + size - 1
        for i in row..<row+size {
            matrix[row][i] = fillValue
            matrix[i][col] = fillValue
            matrix[end][i] = fillValue
            matrix[i][end] = fillValue
        }
    }
    
    func printMatrix( matrix: [[Int]] ) {
        for i in 0..<matrix.count {
            for j in 0..<matrix[0].count {
                print( "\(matrix[i][j]) ", terminator: "" )
            }
            print()
        }
    }
}

var size = 4
let solution = Solution()
let matrix = solution.prettyPrint( &size )
solution.printMatrix(matrix: matrix)
